### ISSUE TITLE
Linewise, characterwise and blockwise modifiers not working

### ISSUE THREAD

**Issue by nmercado1986** [wellle/targets.vim#214]
Vanilla vim supports forcing motions to be performed in a linewise (V), blockwise (C_v) and characterwise(v) fashion. See :help o_v

This isn't working for motions using targets.vim text objects.

**Example scenario:** ( | indicates cursor position in normal mode)

```
|function() {
     someCode;
}
someCode;
```

**Expected:** dVa} results in:
```
|
someCode;
```

**Actual:** dVa} results in:
```
|function()
someCode;
```

Reporting this as I couldn't find an open issue referencing linewise operators besides the description on README.md.

**Comment by nmercado1986** [comment]
In case this is related, when on VISUAL LINE mode, `a}` causes the editor to switch modes to VISUAL, instead of the expected VISUAL LINE selection.

**Comment by wellle** [comment]
@nmercado1986: Thanks for bringing this up, I completely agree with you!

Unfortunately I'm not aware of any way to make it work. Currently Vim is handling `d` and `V` itself, then finds an omapping for `a`, so it targets.vim handles the `a}` part, to select such a block. targets.vim itself can see whether we're in visual or operator-pending mode by invoking `mode()` (see `:h mode()`). I just tested it with `da{` and `dVa{` respectively and I don't see any way inside of the omapping whether there was a `V` or not.

I guess targets.vim could technically create an omapping for `V` itself, but that seems wrong to me. To be honest I feel like Vim/Neovim should actually take care of this: When seeing the (unmapped) `V` in operator pending mode, remember to force linewise. Then invoke any mappings as usual, but before applying the operator, if `V` was pressed, convert the returned selection to be linewise, and only then apply the operator. Because currently (if it were possible) each text object plugin would need to implement this itself.

For example the matchit plugin has the same problem. Try putting your cursor on `{` and type `dV%`. On my machine it leads to the same result where `function()` remains.

And even some vim motions have the problem. Consider `gn`: Search for something, if you type `gn` it selects the next match. If instead you type `dgn` it deletes the next match and that can be repeated with `.`, super useful! But I'd like to see `dVgn` work too, where I'd expect it to delete the full line containing the next match. So with repeating I could do interactively what `:g//d` does, kinda.

(Even more off topic, but somewhat related, if you go straight into visual line mode and use a "short" text object Vim will switch to visual character mode. For example from normal mode in your example with cursor on `{` (without targets.vim) `Va{` will select the block character wise. So in the line/character-ness of visual mode doesn't really get respected when using text objects. At the same time this is convenient when using "big" text objects like `vip`, where it automatically switches to linewise mode. All things considered the current way is probably best, but it does lead to cases where `Va{d` and `dVa{` behave differently already)

Anyway, to sum it up, I'd love to support this properly, but I believe it should be either fixed directly in Vim/Neovim, or Vim/Neovim would need to offer me a way to recognize that the selection should be forced linewise (maybe extend `mode()` again?).

**Comment by nmercado1986** [comment]
Thanks so much for referencing this issue in the neovim repo.
Could I ask you add a similar request it in the original vim repo? 
I really hope Bram will accept this behavior as inconsistent in Vim and could be improved.
I know I could do this myself but I wouldn't have any alternatives but to plagiarize your note in the neovim repo, which is very well-explained as you are clearly more familiarized than me with the text-object operations implementation in Vim and Neovim, and their consequences for plugin developers.
In any case, thanks for your help so far.

**Comment by wellle** [comment]
@nmercado1986: I'll do that 👍

**Comment by wellle** [comment]
@nmercado1986: I opened an issue and got some response already. The issue is known and there already exists a patch for it. It's just that they couldn't find a proper name for the new flag yet. Not because of disagreement, but apparently just because of lack of activity.

See https://github.com/vim/vim/issues/3490#issuecomment-425276862 and the links in there.

**Comment by wellle** [comment]
Quick update: The vim issue https://github.com/vim/vim/issues/3490 has been closed and a solution has been released in https://github.com/vim/vim/commit/5976f8ff00efcb3e155a89346e44f2ad43d2405a as patch 8.1.0648. 🎉 

Hopefully we'll see the same solution added to Neovim too. In the meantime I might update targets.vim to use that new feature.

**Comment by dsummersl** [comment]
I notice that docs still mentions that force linewise is documented as not working; and it doesn't work for me in neovim.
